File: client/iih_stub/iih.ci_stub/iih.ci.ord/cior/d/CiordrptbttestAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.ord.cior.d
{
	/// <summary>
	/// 备血检验结果DO
	/// </summary>
	public class CiordrptbttestAggDO : BaseAggDO
	{	
	    private static string CIORDBTTESTDO_CIORDBTTESTITMDO="iih.ci.ord.cior.d.CiOrdBtTestItmDO";

        public CiordrptbttestAggDO() {
            this.setParentDO(new CiOrdBtTestDO());
        }

  	    public CiOrdBtTestDO getParentDO() {
		    return ((CiOrdBtTestDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(CiOrdBtTestDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public CiOrdBtTestItmDO[] getCiOrdBtTestItmDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(CIORDBTTESTDO_CIORDBTTESTITMDO));
		    if (dos == null || dos.Length==0){return null;}
            CiOrdBtTestItmDO[] r = new CiOrdBtTestItmDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as CiOrdBtTestItmDO;
            }
	        return r;
		    
	    }
	    
	    public void setCiOrdBtTestItmDO(CiOrdBtTestItmDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new CiOrdBtTestDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (CIORDBTTESTDO_CIORDBTTESTITMDO == clzName)
	        {
                return new CiOrdBtTestItmDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.ord.cior.d.CiOrdBtTestItmDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/cior/d/CiordrptbttestAggDO.java
package iih.ci.ord.cior.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.ord.cior.d.desc.CiordrptbttestAggDODesc;



/**
 * 备血检验结果
 */
public class CiordrptbttestAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public CiOrdBtTestDO getParentDO() {
		return ((CiOrdBtTestDO) super.getParentDO());
	}

	public void setParentDO(CiOrdBtTestDO headDO) {
		setParent(headDO);
	}

	public CiOrdBtTestItmDO[] getCiOrdBtTestItmDO() {
		IBaseDO[] dos = getChildren(CiOrdBtTestItmDO.class);
		if(dos==null) return null;
		CiOrdBtTestItmDO[] result=new CiOrdBtTestItmDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(CiOrdBtTestItmDO)dos[i];
		}
		return result;
	}
	
	public void setCiOrdBtTestItmDO(CiOrdBtTestItmDO[] dos) {
		setChildren(CiOrdBtTestItmDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		CiordrptbttestAggDODesc desc = new CiordrptbttestAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new CiOrdBtTestDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.ord.cior.d.CiOrdBtTestItmDO")) {
             return new CiOrdBtTestItmDO();
         }
         return null;
     }
}